frontends/web/src/routes/exchange/bitrefill.tsx

```diff
@@ -0,0 +1,251 @@
+export const localeMapping: Readonly<Record<string, string>> = {
+  en: 'en',
+  de: 'de',
+  fr: 'fr',
+  es: 'es',
+  it: 'it',
+  pt: 'pt',
+  ja: 'ja',
+  zh: 'zh-Hans'
+};
+
+// Map coins supported by Bitrefill
+const coinMapping: Readonly<Record<string, string>> = {
+  btc: 'bitcoin',
+  tbtc: 'bitcoin',
+  ltc: 'litecoin',
+  eth: 'ethereum',
+  usdt: 'usdt_erc20',
+  usdc: 'usdc_erc20',
+};
+
+type TProps = {
+  accounts: IAccount[];
+  code: AccountCode;
+};
+
+const getURLOrigin = (uri: string): string | null => {
+  try {
+    return new URL(uri).origin;
+  } catch (e) {
+    return null;
+  }
+};
+
+export const Bitrefill = ({ accounts, code }: TProps) => {
+  const { t } = useTranslation();
+  const { isDarkMode } = useDarkmode();
+  const account = findAccount(accounts, code);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+  const [height, setHeight] = useState(0);
+  const resizeTimerID = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const bitrefillInfo = useLoad(() => getBitrefillInfo('spend', code));
+
+  const config = useLoad(getConfig);
+  const [agreedTerms, setAgreedTerms] = useState(false);
+
+  const hasOnlyBTCAccounts = accounts.every(({ coinCode }) => isBitcoinOnly(coinCode));
+
+  useEffect(() => {
+    if (config) {
+      setAgreedTerms(config.frontend.skipBitrefillWidgetDisclaimer);
+    }
+  }, [config]);
+
+  const onResize = useCallback(() => {
+    if (resizeTimerID.current) {
+      clearTimeout(resizeTimerID.current);
+    }
+    resizeTimerID.current = setTimeout(() => {
+      if (containerRef.current) {
+        setHeight(containerRef.current.offsetHeight);
+      }
+    }, 200);
+  }, []);
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+      if (resizeTimerID.current) {
+        clearTimeout(resizeTimerID.current);
+      }
+    };
+  }, [onResize]);
+
+  const handleMessage = useCallback(async (event: MessageEvent) => {
+    if (
+      !account
+      || !bitrefillInfo?.success
+      || ![getURLOrigin(bitrefillInfo.url), 'https://embed.bitrefill.com'].includes(event.origin)
+    ) {
+      return;
+    }
+
+    const data = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+
+    switch (data.event) {
+    case 'request-configuration': {
+      event.source?.postMessage({
+        event: 'configuration',
+        ref: bitrefillInfo.ref,
+        utm_source: 'BITBOX',
+        theme: isDarkMode ? 'dark' : 'light',
+        hl: i18n.resolvedLanguage ? localeMapping[i18n.resolvedLanguage] : 'en',
+        paymentMethods: account.coinCode ? coinMapping[account.coinCode] : 'bitcoin',
+        refundAddress: bitrefillInfo.address,
+        // Option to keep pending payment information longer in session, defaults to 'false'
```

Comment:
could you add a link to their documentation somewhere (in this file or in commit message)?